src/app/modules/core/core.errors.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * A custom base error class that encapsulates the name, message, status code,
+ * and logging meta string (if any) for the error.
+ */
+export class ApplicationError extends Error {
+  /**
+   * Http status code for the error to be returned in the response.
+   */
+  status: number
+  /**
+   * Meta string to be logged by the application logger, if any.
+   */
+  meta?: string
+
+  constructor(message?: string, status?: number, meta?: string) {
+    super()
+
+    Error.captureStackTrace(this, this.constructor)
+
+    this.name = this.constructor.name
+
+    this.message = message || 'Something went wrong. Please try again.'
```

Comment:
idea: I'll upgrade us to ts4 then you can use `??=`

Reply:
Not for this, since this is still an assignment. Technically I can already use `message ?? 'Something went wrong...'` woops